### PR TITLE
Adding Namespace to Smoke Test

### PR DESCRIPTION
**Pull request by BrandonDalton**
#### What this PR does

Adding in namespace field for smoke-test-job.yaml

#### Which issue(s) this PR fixes or relates to #2529 

I have been implementing Mimir OSS within an organization however due to security I am unable to access or deploy things to default namespace. This will allow to deploy it in the same namespace alongside the rest of the Grafana Mimir components and allow us to deploy this test and interact with it.

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by BrandonDalton** [comment]
Hello 👋 

Would you like me to open an issue for such a small issue I have encountered?

Cheers :)

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [ENHANCEMENT] Add `NOTES.txt` to show endpoints URLs for the user at install/upgrade. #2189
 * [ENHANCEMENT] Add ServiceMonitor for overrides-exporter. #2068
 * [ENHANCEMENT] Add `nginx.resolver` for allow custom resolver in nginx configuration and `nginx.extraContainers` which allow add side containers to the nginx deployment #2196
-
+* [ENHANCEMENT] Add `namespace` to smoke-test helm template to allow the job to be deployed within the same namespace as the rest of the deployment.
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:70]:
```suggestion
* [ENHANCEMENT] Add `namespace` to smoke-test helm template to allow the job to be deployed within the same namespace as the rest of the deployment. #2515

```

File: operations/helm/charts/mimir-distributed/templates/smoke-test/smoke-test-job.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@ metadata:
     {{- toYaml .Values.smoke_test.annotations | nindent 4 }}
     {{- end }}
     "helm.sh/hook": test
+  namespace: {{ .Release.Namespace | quote }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/smoke-test/smoke-test-job.yaml:13]: I think you also need to run `make build-helm-tests` from the root of the repo.

**Review by dimitarvdimitrov — COMMENTED** [review]
Thank you for the contribution :)

File: operations/helm/charts/mimir-distributed/templates/smoke-test/smoke-test-job.yaml (hunk 2 above)
Comment (BrandonDalton) [reply operations/helm/charts/mimir-distributed/templates/smoke-test/smoke-test-job.yaml:13]: Done :)

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM. @krajorama can you also take a look and merge this?

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 3]
```diff
@@ -67,6 +67,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [ENHANCEMENT] Add `NOTES.txt` to show endpoints URLs for the user at install/upgrade. #2189
 * [ENHANCEMENT] Add ServiceMonitor for overrides-exporter. #2068
 * [ENHANCEMENT] Add `nginx.resolver` for allow custom resolver in nginx configuration and `nginx.extraContainers` which allow add side containers to the nginx deployment #2196
+* [ENHANCEMENT] Add `namespace` to smoke-test helm template to allow the job to be deployed within the same namespace as the rest of the deployment. #2515
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:70]: Please rebase to current main and move this entry to the `main / unreleased` section. 3.0.0 is already released.

**Review by krajorama — CHANGES_REQUESTED** [review]

File: operations/helm/charts/mimir-distributed/CHANGELOG.md (hunk 3 above)
Comment (BrandonDalton) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md]: Thanks, 

Done :)

**Review by krajorama — APPROVED** [review]

**Pull request merged**